performance_regression_test.py

```diff
@@ -891,3 +891,65 @@ def test_latency_write_with_upgrade(self):
     def test_latency_mixed_with_upgrade(self):
         self._prepare_latency_with_upgrade()
         self.run_workload_and_upgrade(stress_cmd=self.params.get('stress_cmd_m'))
+
+
+class PerformanceRegressionMaterializedViewLatencyTest(PerformanceRegressionTest):
+
+    def test_read_mv_latency(self):
+        # next 3 lines, is a workaround to have it working inside `latency_calculator_decorator`
+        self.cluster = self.db_cluster  # pylint: disable=attribute-defined-outside-init
+        self.tester = self  # pylint: disable=attribute-defined-outside-init
+        self.monitoring_set = self.monitors  # pylint: disable=attribute-defined-outside-init
+
+        self.run_fstrim_on_all_db_nodes()
+        self.preload_data() #prepare_write_cmd
+        self.wait_no_compactions_running()
+        self.run_fstrim_on_all_db_nodes()
+
+        self.create_test_stats(sub_type="read", append_sub_test_to_name=False, test_index="mv-overloading-latency-read")
+        self.run_stress_thread(stress_cmd=self.params.get('stress_cmd_r'), stress_num=1,
+                                              stats_aggregate_cmds=False)
+
+        self.steady_state_read_workload_latency()  #stress_cmd_r
+        self.do_rewrite_workload() #stress_cmd_no_mv + #stress_cmd_r
+        self.wait_mv_sync() #stress_cmd_r
+        self.do_rewrite_workload_with_mv() #stress_cmd_mv + #stress_cmd_r
+        self.loaders.kill_stress_thread()
+        #self.check_latency_during_ops() TODO: need to implement custom check in case of running test on regular basis
+
+
+    @latency_calculator_decorator
+    def steady_state_read_workload_latency(self):
+        InfoEvent(message='start_read_workload_latency begin').publish()
+        time.sleep(15*60)
+        InfoEvent(message='start_read_workload_latency ended').publish()
+
+    @latency_calculator_decorator
+    def do_rewrite_workload(self):
+        base_cmd = self.params.get('stress_cmd_no_mv')
+        stress_queue = self.run_stress_thread(stress_cmd=base_cmd, stress_num=1, stats_aggregate_cmds=False)
+        results = self.get_stress_results(queue=stress_queue, store_results=False)
+        self.display_results(results, test_name='do_rewrite_workload')
+
+    @latency_calculator_decorator
+    def wait_mv_sync(self):
+        node_count = len(self.db_cluster.nodes)
+        node1 = self.db_cluster.nodes[0]
+        node1.run_cqlsh("CREATE TABLE IF NOT EXISTS scylla_bench.test (pk bigint,ck bigint,v blob,PRIMARY KEY(pk, ck)) WITH compression = { }")
+        node1.run_cqlsh("CREATE MATERIALIZED VIEW IF NOT EXISTS scylla_bench.view_test AS SELECT * FROM scylla_bench.test where v IS NOT NULL AND ck IS NOT NULL AND pk IS NOT NULL PRIMARY KEY (v, pk, ck)")
+        start_time = time.time()
+        while True:
```

Comment:
we have `sdcm.utils.nemesis_utils.indexes.wait_for_view_to_be_built` for waiting for index to be built